pkg/koordlet/qosmanager/config/config.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"flag"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/runtime"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/featuregate"
+
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/qosmanager/plugins"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/qosmanager/plugins/cpuburst"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/qosmanager/plugins/cpuevict"
+	"github.com/koordinator-sh/koordinator/pkg/koordlet/qosmanager/plugins/cpusuppress"
+)
+
+var (
+	DefaultMutableQoSManagerFG featuregate.MutableFeatureGate = featuregate.NewFeatureGate()
+	DefaultQoSManagerFG        featuregate.FeatureGate        = DefaultMutableQoSManagerFG
+
+	cpuSuppressFeature = new(cpusuppress.CPUSuppress).Feature()
+	cpuBurstFeature    = new(cpuburst.CPUBurst).Feature()
+	cpuEvictFeature    = new(cpuevict.CPUEvictor).Feature()
+
+	defaultQoSManagerFG = map[featuregate.Feature]featuregate.FeatureSpec{
+		cpuSuppressFeature: {Default: false, PreRelease: featuregate.Alpha},
+		cpuBurstFeature:    {Default: false, PreRelease: featuregate.Alpha},
+		cpuEvictFeature:    {Default: false, PreRelease: featuregate.Alpha},
+	}
+
+	QoSPluginFactories = map[featuregate.Feature]plugins.PluginFactoryFn{
+		cpuSuppressFeature: cpusuppress.New,
+		cpuBurstFeature:    cpuburst.New,
+		cpuEvictFeature:    cpuevict.New,
+	}
+)
+
+type Config struct {
+	FeatureGates       map[string]bool
+	PluginExtraConfigs map[string]string
```

Comment:
how about we keep the config simple at first, support plugin extra config latter.
in the future, maybe we define as map[string]interface{}, extra plugins can register their own config.